File: python/989_数组形式的整数加法_easy.py
class Solution:
    def addToArrayForm(self, A: List[int], K: int) -> List[int]:
        n = len(A)
        ans = []
        for i in range(n-1, -1, -1):
            A[i] += K%10
            K //= 10
            if(A[i]>=10):
                A[i] -= 10
                K += 1
            ans.append(A[i])
        while(K>0):
            ans.append(K%10)
            K //= 10
        res = []
        for i in range(len(ans)-1, -1, -1):
            res.append(ans[i])
        return res

File: cpp/989_数组形式的整数加法_easy.cpp
class Solution {
public:
    vector<int> addToArrayForm(vector<int>& A, int K) {
        int n = A.size();
        vector<int> ans;
        for(int i=n-1; i>=0; i--){
            A[i] += K%10;
            K /= 10;
            if(A[i]>=10){
                A[i] -= 10;
                K++;
            }
            ans.push_back(A[i]);
        }
        for(; K>0; K/=10)
            ans.push_back(K%10);
        reverse(ans.begin(), ans.end());
        return ans;
    }
};